pkg/sqlite/gallery.go

```diff
@@ -787,23 +800,58 @@ func (qb *GalleryStore) makeQuery(ctx context.Context, galleryFilter *models.Gal
 	return &query, nil
 }
 
-func (qb *GalleryStore) Query(ctx context.Context, galleryFilter *models.GalleryFilterType, findFilter *models.FindFilterType) ([]*models.Gallery, int, error) {
-	query, err := qb.makeQuery(ctx, galleryFilter, findFilter)
+func (qb *GalleryStore) Query(ctx context.Context, options models.GalleryQueryOptions) (*models.GalleryQueryResult, error) {
```

Comment:
Align on how it work for Scene to be able use the exact same logic.
Multiple changes are just a side of this.

Alternative would've been to create an additional `QueryWithOptions` method, which would end up of having 2 ways to do the same thing.